_posts/2019-09-02-shorter-isn't-better.md

```diff
@@ -0,0 +1,256 @@
+
+    ```Javascript
+    const pfGetBalanceInFC 
+        // = (balance = 0) => after(after(cGetFC))(cGetBalance)(balance);
+        = after(after(cGetFC))(cGetBalance);
+    ```
+
+So the point-free style `getBalanceInFC` is just `after(after(cGetFC))(cGetBalance)`. It's short and abstract but it's totally confusing! Even though the point free style does has its advatages, if we extract the pattern above into a combinator:
+
+```Javascript
+const owl 
+    // = f => g => after(after(f))(g);
+    = after(after)(after); // applying same rules above
+```
+
+and its Haskell version looks much more terrifying:
+
+```Haskell
+-- because it looks like an owl
+owl = (.).(.)
+```
+
+Then the `owl` works for any number of parameters versions of function `f` given any 2 parameter function like `getBalance` that has the same amount of parameters:
+
+```Javascript
+const f = (x, y, z) => x + y + z;
+const g = (x, y) => x / y;
+const p = (w, x, y, z) => f((g(w, x)), y, z);
+const owlP = owl(curry(f))(curry(g)); 
+
+console.log(p(1,2,3,4) === owlP(1)(2)(3)(4)) //true
+```
+Which means given same parameters, the execution result of `p` and `owlP` is always the same (one can prove the feature by the type signature of our `after` combinator). This does provide maintanability but still, hard to read and understand.
+
+## Conclusion
+
+Coding practice is a kind of social practice because it involves cooperations and communications. Communications require certain education background and intelligence between the participants. Here is an example when it comes to communications: Whenever it comes to a situation that people ask me to explain the idea of `monad`, the shortest and the best version is always the version comes from `James Iry`, from his highly entertaining [Brief, Incomplete and Mostly Wrong History of Programming Languages](http://james-iry.blogspot.com/2009/05/brief-incomplete-and-mostly-wrong.html):
+
+> A monad is just a monoid in the category of endofunctors
+
+I believe whenever after a person hearing this, the conversation ends. Because in order to understand the meaning of the sentence, the listener needs to understand a bunch of mathematical ideas like: `poset` and `monoid`, `category` connection between `category` and `poset`, `galois connection` and generalize it to `adjoint situation`, `functor` and `natrual transformations`, and `Kleisli Categories`, which if he does he won't even ask in the first place. But if I start with a simple commutative diagram, which I thought is the best part of the `category theory` that you can prove things by drawing strings and dots, and then proceed with the idea of `Kleisli Categories`, they might gain some understand about the intimidating idea.
```

Comment:
I believe the conversation would end when most people hear this, because to understand the sentence, they need to understand a bunch of mathematical ideas like poset, monoid, category, and endofunctors, which would require knowledge of Galois connection, adjoint situation, fuctor, natural transformations, and Kleisli Categories. But if I started with simple a commutative diagram, which in my opinion is the best section of the category theory, I can walk them through the proof by just drawing lines and dots, then proceed to explain the concept of Kleisli Categories, then perhaps they can gain some understanding of these complex concepts.